src/test/java/com/diffblue/deeptestutils/regression/ReflectorGetInstanceTest.java

```diff
@@ -1,20 +1,164 @@
+        "com.diffblue.deeptestutils.regression.AbstractClass");
+    int retval = ac.concreteMethod();
+    Assert.assertEquals(retval, 23);
+  }
+
+  // When we call Reflector.getInstance on an abstract class and call concrete
+  // methods on it that change its state, as long as no abstract methods are
+  // also called it behaves just as a concrete class would.
+  @Test
+  public void abstractClassConcreteMutators() throws InvocationTargetException {
+    AbstractClass ac = (AbstractClass) Reflector.getInstance(
+        "com.diffblue.deeptestutils.regression.AbstractClass");
+    ac.setField("");
+    ac.appendToField("test");
+    ac.appendToField("cover");
+    Assert.assertEquals(ac.getField(), "testcover");
+  }
+
+  // Just like in the case of concrete classes, calling Reflector.getInstance on
+  // an abstract class does not cause a call to the constructor of that class.
+  // If the constructor was called, the value of the field would be 100.
+  @Test
+  public void abstractClassNoConstructorCall()
+      throws InvocationTargetException {
+    AbstractClassWithConstructor ac =
+        (AbstractClassWithConstructor) Reflector.getInstance(
+            "com.diffblue.deeptestutils.regression."
+                + "AbstractClassWithConstructor");
+    Assert.assertEquals(ac.getField(), 0);
+  }
+
+  private char unusedInterfaceHelper(Interface in) {
+    if (in == null) {
+      return 'n';
+    }
+    return 'i';
+  }
+
+  // We can create instances of interfaces using Reflector.getInstance. If we do
+  // not call any methods on the instance, it should behave just like an
+  // instance of an implementing class.
+  @Test
+  public void unusedInterface() throws InvocationTargetException {
+    Interface in = (Interface) Reflector.getInstance(
+        "com.diffblue.deeptestutils.regression.Interface");
+    char retval = unusedInterfaceHelper(in);
+    Assert.assertEquals(retval, 'i');
+  }
+
+  // When we call Reflector.getInstance on an interface and try to call a
+  // (non-default) method on it, an AbstractMethodError is thrown.
+  // TODO Add a test for default methods of interfaces, which should only be run
+  // TODO using Java 8, as default methods were a new feature in Java 8.
+  @Test
+  public void interfaceMethod() throws InvocationTargetException {
+    thrown.expect(AbstractMethodError.class);
+    Interface in = (Interface) Reflector.getInstance(
+        "com.diffblue.deeptestutils.regression.Interface");
+    in.method();
+  }
+
+  // When we call Reflector.getInstance on a concrete class whose static
+  // initialiser throws an exception, it wraps this exception within an
+  // InvocationTargetException.
+  @Test
+  public void concreteClassWithBadStaticInit() throws Throwable {
+    thrown.expect(NullPointerException.class);
+    try {
+      Reflector.getInstance(
+          "com.diffblue.deeptestutils.regression.BadStaticInit");
+    } catch (InvocationTargetException e) {
+      throw e.getCause();
```

Comment:
:no_entry_sign: I think better to add an assert the cause is an NPE since currently this test passes if an NPE is thrown and not wrapped

Reply:
Good point. I changed the test as you suggested and also added an `Assert.assertTrue(false);` in the `try` block to show the `try` never succeeds.
Worth noting that I'm pretty sure that the tests we currently generate look like the original version of this regression test, i.e. they will say `thrown.expect(SomeException.class);`, catch the InvocationTargetException and throw its cause. Maybe we should change this so the tests generated by Diffblue Cover look more like the new version of this regression test?